qtaf-core/src/main/java/de/qytera/qtaf/core/event_subscriber/step/StepLoggerSubscriber.java

```diff
@@ -278,39 +279,55 @@ private File saveStepScreenshot(File srcFile, String path) {
+            String stepMessage = "[Step] [%s] [%s] %s";
+
+            logger.info(
+                    stepMessage.formatted(
+                            stepExecutionInfo.getId(),
+                            stepExecutionInfo.getAnnotation().name(),
+                            message
+                    )
+            );
 
-        if (stepExecutionInfo.getLogMessage() != null) {
-            for (AssertionLogMessage m : stepExecutionInfo.getLogMessage().getAssertions()) {
-                if (m.hasFailed()) {
-                    logger.info(
-                            assertionMessage.formatted(
-                                    stepExecutionInfo.getId(),
-                                    m.type(),
-                                    ConsoleColors.red("failed"),
-                                    m.getMessage()
-                            )
-                    );
-                } else {
-                    logger.info(
-                            assertionMessage.formatted(
-                                    stepExecutionInfo.getId(),
-                                    m.type(),
-                                    ConsoleColors.green("passed"),
-                                    m.getMessage()
-                            )
-                    );
+            if (stepExecutionInfo.getLogMessage() != null) {
+                for (AssertionLogMessage m : stepExecutionInfo.getLogMessage().getAssertions()) {
+                    if (m.hasFailed()) {
+                        logger.info(
+                                assertionMessage.formatted(
+                                        stepExecutionInfo.getId(),
+                                        m.type(),
+                                        ConsoleColors.red("failed"),
+                                        m.getMessage()
+                                )
+                        );
+                    } else {
+                        logger.info(
+                                assertionMessage.formatted(
+                                        stepExecutionInfo.getId(),
+                                        m.type(),
+                                        ConsoleColors.green("passed"),
+                                        m.getMessage()
+                                )
+                        );
+                    }
                 }
             }
         }
     }
+
+    /**
+     * The QTAF-User has the option to configure
+     * if the test steps should get logged to the console.
+     * If in the qtaf.json the login.logSteps is set to false
+     * no steps should get logged.
+     * This is usefully if the user wants shorter logs.
+     *
+     * @return boolean that is to true if logging is wanted and to false if logging is unwanted
+     */
+    private boolean checkIfStepLoggingIsConfigured(){
+        ConfigMap config = QtafFactory.getConfiguration();
+        return config.getBoolean("logging.logSteps");
```

Comment:
You should use the "getBoolean" method with a default value. Otherwise users will get an exception if they don't have set the configuration value

![image](https://github.com/Qytera-Gmbh/QTAF/assets/90603395/cf1a0438-6b91-4203-a1d6-65d393c59dbf)

Reply:
Also you should use the "isStepLoggingEnabled" of the "QtafTestExecutionConfigHelper" class and remove the method in this class.

Reply:
Default value added in https://github.com/Qytera-Gmbh/QTAF/pull/283/commits/85bdd3ab8135426faab1142cca66e13cadd7cf68.